### DISCUSSION TITLE
Buffer delete

### DISCUSSION THREAD

**Discussion by dpetka2001** [ibhagwan/fzf-lua#1292] [Q&A] [answered]
Hi, there. So, I would like to add an action to `buffers` picker to delete a buffer. I tried the following
```lua
      opts.buffers = vim.tbl_deep_extend("force", opts.buffers, {
        actions = {
          ["ctrl-d"] = actions.buf_del,
        },
      })
```
and it works in that it deletes the buffer that I choose, but I would like, if it's possible, for the window to not close and I can continue working with the `buffers` picker. Is there some way for the keymap above to still work without closing the `buffers` picker after the action takes place?

**Comment by dpetka2001** [comment, marked as answer]
Ok, I should really read the README more carefully before asking questions like these. 
```lua
      opts.buffers = vim.tbl_deep_extend("force", opts.buffers, {
        actions = {
          ["ctrl-d"] = { fn = actions.buf_del, reload = true },
        },
      })
```
did the trick.

**Reply by ibhagwan** [reply]
I would also add `['ctrl-x']=false` to disable the default.

**Reply by dpetka2001** [reply]
:man_facepalming: Default works just fine for me. I just didn't read carefully **again**. No need for new keymap. 

Thank you as always for your time and advice to new users :smile:
